lib/statsd/instrument/aggregator.rb

```diff
@@ -0,0 +1,246 @@
+# frozen_string_literal: true
+
+module StatsD
+  module Instrument
+    class Aggregator
+      CONST_SAMPLE_RATE = 1.0
+      COUNT = :c
+      DISTRIBUTION = :d
+      MEASURE = :ms
+      HISTOGRAM = :h
+      GAUGE = :g
+
+      class << self
+        def finalize(aggregation_state, sink, datagram_builders, datagram_builder_class)
+          proc do
+            aggregation_state.each do |_key, counter|
+              if datagram_builders[counter[:no_prefix]].nil?
+                datagram_builders[counter[:no_prefix]] =
+                  create_datagram_builder(datagram_builder_class, no_prefix: counter[:no_prefix])
+              end
+              sink << datagram_builders[counter[:no_prefix]].c(
+                counter[:name],
+                counter[:value],
+                CONST_SAMPLE_RATE,
+                counter[:tags],
+              )
+            end
+            aggregation_state.clear
```

Comment:
```suggestion
            aggregation_state.each do |_key, counter|
              datagram_builders[counter[:no_prefix]] ||= datagram_builder_class.new(
                prefix: no_prefix ? nil : @metric_prefix,
                default_tags: @default_tags,
              )
              sink << datagram_builders[counter[:no_prefix]].c(
                counter[:name],
                counter[:value],
                CONST_SAMPLE_RATE,
                counter[:tags],
              )
            end
```

You could just inline this one location where we call the `create_datagram_builder` method (unless you were trying to silence rubocop, or something).

I'd also drop the `aggregation_state.clear` since this is running in the context of a finalizer - and everything is (theoretically) about to be garbage collected anyways.